samples/json-update-data.js

```diff
@@ -0,0 +1,137 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+async function updateWithJsonData(instanceId, databaseId, projectId) {
+  // [START spanner_update_data_with_json_column]
+  // Imports the Google Cloud client library.
+  const {Spanner} = require('@google-cloud/spanner');
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = 'my-project-id';
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+
+  // Creates a client.
+  const spanner = new Spanner({
+    projectId: projectId,
+  });
+
+  // Gets a reference to a Cloud Spanner instance and database.
+  const instance = spanner.instance(instanceId);
+  const database = instance.database(databaseId);
+
+  // Instantiate Spanner table objects.
+  const venuesTable = database.table('Venues');
+
+  const data = [
+    {
+      VenueId: '19',
+      VenueDetails: {rating: 9, description: 'This is a nice place'},
+      LastUpdateTime: 'spanner.commit_timestamp()',
+    },
+    {
+      VenueId: '4',
+      // VenueDetails must be specified as a string, as it contains a top-level
+      // array of objects that should be inserted into a JSON column. If we were
+      // to specify this value as an array instead of a string, the client
+      // library would encode this value as ARRAY<JSON> instead of JSON.
+      VenueDetails: `[
```

Comment:
@zoercai Here's an example of the issue with top-level JSON arrays. The client library does not know whether this should be encoded as `JSON` or `ARRAY<JSON>` if it is specified as an actual array instead of a string. Specifying it as a string will cause the client library to just encode it as a string, which is the default encoding for `JSON` values.